CookingBook/src/networking/Client/Client.cpp

```diff
@@ -0,0 +1,218 @@
+	LogInfo("Size of the file "+sFileName+" was received:"+std::to_string(iRecvBufferSize));
+	char *Buffer = new char[iRecvBufferSize];
+
+	std::ofstream File;
+	File.open(sFileName, std::ios::binary);
+	int iTmpRecvSize = iRecvBufferSize;
+	int iCount = 0;
+	while(iTmpRecvSize>0){
+		iCount = recv(ClientSocket,Buffer,iRecvBufferSize,0);
+		if(iCount==-1)break;
+		iTmpRecvSize-=iCount;
+	}
+	LogInfo("File input was received from server");
+	File.write(Buffer,iRecvBufferSize);
+	LogInfo("File was recreated on client successfully");
+	delete[] Buffer;
+	File.close();
+	std::ifstream InputFile;
+	InputFile.open(sFileName,std::ios::binary|std::ios::ate);
+	iTmpRecvSize = InputFile.tellg();
+	std::cout<<"file size = "<<iTmpRecvSize<<" received size = "<<iRecvBufferSize<<std::endl;
+	if(iTmpRecvSize!=iRecvBufferSize){
+		send(ClientSocket,"file",strlen("file"),0);
+		RecvFile(sFileName);
+	}
+}
+
+void Client::Init() {
+	assert(!WSAStartup(MAKEWORD(2, 2), &WSA)&&"Couldn`t WSA on client");
+	LogInfo("WSA started on client");
+	
+	ClientInfo.sin_addr.S_un.S_addr = inet_addr(sIpAddress.c_str());
+	ClientInfo.sin_port = htons(iPort);
+	ClientInfo.sin_family = AF_INET;
+	assert(!((ClientSocket = socket(AF_INET , SOCK_STREAM, 0))==SOCKET_ERROR)&&"Couldn`t create socket on client");
+
+	LogInfo("Client socket was created");
+}
+//this function is just for the test of server, the idea is that we will have various functions
+// for different requests, so server will handle them
+void Client::ConnectToServer() {
+	if(TryConnection()){
+		while (true) {
+			std::cout << "Enter message" << std::endl;
+			std::cin >> szMainBuffer;
+			send(ClientSocket, szMainBuffer, strlen(szMainBuffer)+1, 0);
+			LogInfo("Message was sent to server:"+  std::string(szMainBuffer));
+			if (strstr(szMainBuffer, "file")) {
+				std::cout << "Enter file name" << std::endl;
+				std::cin >> szMainBuffer;
+				RecvFile(std::string(szMainBuffer));
+				ZeroMemory(szMainBuffer, sizeof(szMainBuffer));
+			}
+			if(strstr(szMainBuffer,"register")){
+				SUser someuser;
+				someuser.m_sUserName = "dadad";
+				someuser.m_sPassword = "1234";
+				someuser.m_sFavorites = "dada";
+				someuser.m_nID = 1;
+				RegisterUser(someuser);
+
+			}
+			if(strstr(szMainBuffer,"login")){
+				SUser someuser;
+				someuser.m_sUserName = "dadad";
+				someuser.m_sPassword = "djak";
+				someuser.m_sFavorites = "dada";
+				someuser.m_nID = 1;
+				LoginUser(someuser);
+			}
+			else {
+				recv(ClientSocket, szMainBuffer, g_iBufferSize, 0);
+				std::cout << "Message received:" << std::endl;
+				std::cout << szMainBuffer << std::endl;
+				LogInfo("Message was received from server:"+std::string(szMainBuffer));
+			}
+		}
+	}
+	else{
+		std::cout<<"Can`t interact with server"<<std::endl;
+		LogInfo("Couldn`t connect to server");
+	}
+} 
+
+bool Client::RegisterUser(SUser& User){
+	bool bResult;
+	char ServerRespond[g_iMessageSize];
+	send(ClientSocket,reinterpret_cast<char*>(&User),sizeof(User),0);
```

Comment:
You cast user to char*. Maybe it works ok , but looks strange. Maybe it is better have some public get function int user class to get this string or char*